### PR TITLE
Add domain variable

### PR DESCRIPTION
**Pull request by SimmyD**
First attempt at this.  Happy to modify.

Have added the domainUrl value that will default to the connectionOptions.host as previous but when you select custom url's it gives you the option of modifying this as for example the docker defaults for jitsi are meet.jitsi not the url of the hosted platform.

### CODE REVIEW

**Comment by SimmyD** [comment]
fixes #26

File: jitsirtc.js [hunk 1]
```diff
@@ -76,19 +76,21 @@ class JitsiRTCClient extends WebRTCInterface {
...
+          domainUrl = game.settings.get('jitsirtc', 'domainUrl');
           mucUrl = game.settings.get('jitsirtc', 'mucUrl');
           focusUrl = game.settings.get('jitsirtc', 'focusUrl');
           boshUrl = game.settings.get('jitsirtc', 'boshUrl');
         }
 
         options = {
           hosts: {
-            domain: connectionOptions.host,
+            domain: domainUrl,
```
Comment (bekriebel) [inline jitsirtc.js:93]: This is the main URL that is being connected to. If you are setting this to meet.jit.si, then you aren't using a custom server you are connecting to Jitsi's public servers.

**Review by bekriebel — CHANGES_REQUESTED** [review]
Doing this is the same as setting the signaling server URL to the value that you are setting here. So, if you are setting this to `meet.jit.si`, then you aren't connecting to your custom server at all and are using Jitsi's public servers.

File: jitsirtc.js (hunk 1 above)
Comment (SimmyD) [reply jitsirtc.js:93]: Thanks for looking at this.

the domain: isnt actually what is used to connect but bosh: is.  domain: needs to match the XMPP domain used internally on the prosody setup.  In my example i am setting this to meet.jitsi not the public meet.jit.si.  

https://jitsi.github.io/handbook/docs/devops-guide/devops-guide-docker#advanced-configuration

This link shows the XMPP Domain that needs to match the domain:

**Comment by SimmyD** [comment]
"Doing this is the same as setting the signaling server URL to the value that you are setting here. So, if you are setting this to `meet.jit.si`, then you aren't connecting to your custom server at all and are using Jitsi's public servers."

you are right if i set the signalling server URL to meet.jitsi (no meet.jit.si) and set the bosh to the correct (meet.somedomain.com/http-bind) value it works great.

The generated URL used for externalUsersUrl however is generated as https://meet.jitsi/<roomname> instead of https://meet.somedomain.com/<roomname>

so the crux of this is the domain: is actually meant to match the XMPP domain used and bosh: is whats actually used to connect.

Hope this makes sense.

**Comment by bekriebel** [comment]
Thanks for the clarification. Ugh, okay - I see what they are doing. I don't like the way they are doing that in the docker image, but I understand the problem now.

I'm going to hold on merging this for the moment since the updates to support FVTT 0.7.2 refactor the code a lot and this likely won't apply cleanly. I'll rebase it on those changes once I have them done though and will include this, or some form of it, in an upcoming release. Thanks!

**Comment by SimmyD** [comment]
No problem @bekriebel,  Would it be worth putting a note in the readme for people trying to to use the default docker variables?  would save them time working this out.

Thanks again for working on this module.  My group of mates all value it :)

**Comment by bekriebel** [comment]
This shouldn't be needed any more since we now read the config from the designated server. Regardless, if it is needed, the new release also adds this setting as requested.

**Pull request closed without merging**